apps/aecore/test/aecore_tx_test.exs

```diff
@@ -137,4 +137,34 @@ defmodule AecoreTxTest do
     # the nonce is small or equal to account nonce, so the transaction is invalid 
     assert Chain.chain_state().accounts[Wallet.get_public_key()].balance == 100
   end
+
+  test "sum of amount and fee more than balance", tx do
+    sender = Wallet.get_public_key()
+    acc1 = Wallet.get_public_key("M/1")
+    acc2 = Wallet.get_public_key("M/2")
+    amount = 80
+    fee = 40
+    :ok = Miner.mine_sync_block_to_chain()
+    :ok = Miner.mine_sync_block_to_chain()
+    # Send tokens to the first account, sender has 200 tokens
+    payload = %{receiver: acc1, amount: amount}
+    tx_data = DataTx.init(SpendTx, payload, sender, fee, tx.nonce)
+    priv_key = Wallet.get_private_key()
+    {:ok, signed_tx} = SignedTx.sign_tx(tx_data, priv_key)
+    :ok = Pool.add_transaction(signed_tx)
+    :ok = Miner.mine_sync_block_to_chain()
+    # Now acc1 has 80 tokens
+    assert Chain.chain_state().accounts[acc1].balance == 80
+    amount = 50
+    fee = 40
+    # Balance of acc1 is more than amount and fee, send tokens to acc2
+    payload = %{receiver: acc2, amount: amount}
+    tx_data = DataTx.init(SpendTx, payload, acc1, fee, 1)
+    priv_key = Wallet.get_private_key("m/1")
+    {:ok, signed_tx} = SignedTx.sign_tx(tx_data, priv_key)
+    :ok = Pool.add_transaction(signed_tx)
+    :ok = Miner.mine_sync_block_to_chain()
+    # the balance of acc1 and acc2 is not changed because amount + fee > balance of acc1
+    assert Chain.chain_state().accounts[acc2] == nil
```

Comment:
Maybe assert again that acc1 has balance == 80, so that we are sure the transaction was not successful and the balance of acc1 didn't change